### ISSUE TITLE
Unable to create compute instance without specifying networks

### ISSUE THREAD

**Issue by samdbmg** [terraform-providers/terraform-provider-openstack#579]
Hi,

I'm attempting to create a compute instance, and then attach a network to it (this is a minimal example, the actual use case is a module that accepts a list of networks).
The docs at https://www.terraform.io/docs/providers/openstack/r/compute_instance_v2.html don't specify that `network` is required, so I'd assume this is possible, but it doesn't work. However if you comment out the `openstack_compute_interface_attach_v2` and uncomment the `network` block the instance is created successfully. If that's intended behaviour, the documentation should probably reflect it.

Thanks!

### Terraform Version
```
vagrant@ubuntu-xenial$ terraform -v
Terraform v0.11.11
+ provider.openstack v1.13.0
```

### Affected Resource(s)
- openstack_compute_instance_v2
- openstack_compute_interface_attach_v2

### Terraform Configuration Files
```hcl
resource "openstack_compute_instance_v2" "instance" {
  # Instance config
  name        = "net-test"
  image_id    = "e877fbef-3a3f-455a-9d62-3b8637a3ccf5"
  flavor_name = "m1.mini"

#   network {
#     uuid = "e875cfea-295e-44eb-8a7c-e17d56bc05a7"
#   }
}

resource "openstack_compute_interface_attach_v2" "network_attach" {
  instance_id = "${openstack_compute_instance_v2.instance.id}"
  network_id  = "e875cfea-295e-44eb-8a7c-e17d56bc05a7"
}

/*
 * Variables and configuration
 */
variable "openstack_region" {
  description = "OpenStack region to use"
  default     = "RegionOne"
}

/*
 * Example setup
 */
# Use Openstack to deploy this infrastructure
provider "openstack" {
  cloud    = "openstack"
  region   = "${var.openstack_region}"
  insecure = true
  version  = "~> 1.13"
}

```

### Debug Output
https://gist.github.com/samdbmg/ccb3c47ad3739aee3b27a4b97657bd17

### Expected Behavior
The instance to be created, and attached to the network provided.

### Actual Behavior
What actually happened?
```
Error: Error applying plan:

1 error(s) occurred:

* openstack_compute_instance_v2.instance: 1 error(s) occurred:

* openstack_compute_instance_v2.instance: Error creating OpenStack server: Expected HTTP response code [201 202] when accessing [POST https://openstack.mist.rd.bbc.co.uk:8774/v2.1/servers], but got 409 instead
{"conflictingRequest": {"message": "Multiple possible networks found, use a Network ID to be more specific.", "code": 409}}
```
### Steps to Reproduce
Run `terraform apply` on the above example file.

### Important Factoids
OpenStack version is Rocky, otherwise nothing unusual

**Comment by jtopjian** [comment]
Thank you for reporting this. 

> If that's intended behaviour, the documentation should probably reflect it.

Agreed.

We already have a section about [Instances/Servers and Ports](https://www.terraform.io/docs/providers/openstack/r/compute_instance_v2.html#instances-and-ports), but this is another bullet point we need to add.

Using the example you gave:

```hcl
resource "openstack_compute_instance_v2" "instance" {
  # Instance config
  name        = "net-test"
  image_id    = "e877fbef-3a3f-455a-9d62-3b8637a3ccf5"
  flavor_name = "m1.mini"

#   network {
#     uuid = "e875cfea-295e-44eb-8a7c-e17d56bc05a7"
#   }
}

resource "openstack_compute_interface_attach_v2" "network_attach" {
  instance_id = "${openstack_compute_instance_v2.instance.id}"
  network_id  = "e875cfea-295e-44eb-8a7c-e17d56bc05a7"
}
```

Think of these two resources as two separate `openstack` commands, using the `openstack` command-line tool:

```shell
$ openstack server create ...
$ openstack server add port ...
```

You'll get this same error after you execute the first command.

When creating a server/instance without specifying a network, Nova will automatically use a network _if_ the user only has one network available. As soon as there are more than one networks available, Nova returns that error.

Unfortunately Nova has no concept of understanding that you intend to add a network after the server is created :(

**Comment by samdbmg** [comment]
Yes, I had a feeling that might be it; wasn't sure if Terraform/the provider did something clever to coalesce it into one `openstack server create ...`.
I've worked around it myself by specifying the module with `network_id` and `extra_networks_list` input variables (the latter being optional). Of course, when 0.12 is out I can just `foreach` over the network list!

Thanks for having a look.

**Comment by jtopjian** [comment]
Resolved in #669
